corehq/apps/app_manager/static/app_manager/js/manage_releases.js

```diff
@@ -0,0 +1,101 @@
+            self.app = details.app;
+            self.version = details.version;
+            self.location = details.location;
+            self.activatedOn = ko.observable(details.activated_on);
+            self.deactivatedOn = ko.observable(details.deactivated_on);
+            self.errorMessage = ko.observable();
+            self.domId = "restriction_" + self.id;
+            self.ajaxInProgress = ko.observable(false);
+            self.actionText = ko.computed(function () {
+                return (self.active() ? gettext("Remove") : gettext("Add"));
+            });
+            self.toggleStatus = function () {
+                self.active(!self.active());
+            };
+            self.error = ko.observable();
+            self.requestUrl = function () {
+                if (self.active()) {
+                    return initialPageData.reverse('deactivate_release_restriction', self.id);
+                }
+                return initialPageData.reverse('activate_release_restriction', self.id);
+            };
+            self.toggleRestriction = function () {
+                self.ajaxInProgress(true);
+                var oldStatus = self.active();
+                $.ajax({
+                    method: 'POST',
+                    url: self.requestUrl(),
+                    success: function (data) {
+                        if (data.success) {
+                            self.toggleStatus();
+                            self.activatedOn(data.activated_on);
+                            self.deactivatedOn(data.deactivated_on);
+                            self.error(false);
+                        } else {
+                            self.active(oldStatus);
+                            self.errorMessage(data.message);
+                        }
+                    },
+                    error: function () {
+                        self.active(oldStatus);
+                    },
+                    complete: function () {
+                        self.ajaxInProgress(false);
+                        if (self.active() === oldStatus) {
+                            self.error(true);
+                        }
+                    },
+                });
+            };
+            return self;
+        };
+
+        function manageReleasesViewModel(enabledAppReleases) {
+            var self = {};
+            self.enabledAppReleases = ko.observableArray(enabledAppReleases);
+            return self;
+        }
+        var enabledAppReleases = _.map(initialPageData.get('enabled_app_releases'), function (appRelease) {
+            return enabledAppRelease(appRelease);
+        });
+        var viewModel = manageReleasesViewModel(enabledAppReleases);
+        if (enabledAppReleases.length) {
+            $('#managed-releases').koApplyBindings(viewModel);
+        }
+        function manageReleaseSearchViewModel() {
+            var self = {};
+            self.search = function () {
```

Comment:
It's unusual and a bit confusing to be mixing jQuery manipulation and knockout. Can't you create a `searchValue` observable and then set `data_bind='value: searchValue'` when you add the `app-id-search-select` to the crispy layout? And then similarly for the version and location id?

Reply:
made the change but that didn't work. Reverted it for now, will try to add it back.

Reply:
Cool, let me know if you want a second set of eyes on it.